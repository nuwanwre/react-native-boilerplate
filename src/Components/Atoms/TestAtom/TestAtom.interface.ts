export interface ITestAtom {
    filled?: boolean;
}
