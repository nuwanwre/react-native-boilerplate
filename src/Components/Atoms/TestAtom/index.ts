export { TestAtom } from './TestAtom.impl';
