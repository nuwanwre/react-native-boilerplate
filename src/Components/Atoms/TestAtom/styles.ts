import styled from 'styled-components/native';

interface ITestAtomStyled {
    filled?: boolean;
}

export const TestAtomStyled = styled.TouchableOpacity<ITestAtomStyled>`
    border-color: ${({ filled, theme }) =>
        filled ? theme.colors.secondary : 'white'};
    border-width: 1px;
    border-radius: 3px;
    padding: 2px 5px 2px 5px;
`;
