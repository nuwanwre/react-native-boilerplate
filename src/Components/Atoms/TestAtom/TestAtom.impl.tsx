import React from 'react';
import { Text } from 'react-native';

import { ITestAtom } from './TestAtom.interface';

import { TestAtomStyled } from './styles';

export const TestAtom: React.FC<ITestAtom> = ({ filled }) => {
    return (
        <TestAtomStyled filled={filled}>
            <Text>Test Button</Text>
        </TestAtomStyled>
    );
};
