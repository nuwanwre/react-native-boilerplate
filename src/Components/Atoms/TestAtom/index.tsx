import React from "react";
import { TouchableOpacity, Text } from "react-native";

import { TestButtonStyled } from "./styles";

export const TestAtom: React.FC = () => {
    return (
        <TestButtonStyled>
            <Text>Test Button</Text>
        </TestButtonStyled>
    )
}