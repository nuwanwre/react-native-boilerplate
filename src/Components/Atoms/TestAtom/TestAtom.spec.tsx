import React from 'react';

import { shalloWithTheme } from '@Test/Helpers/render';
// import renderer from 'react-test-renderer';

import { TestAtom } from './TestAtom.impl';

describe('Components', () => {
    describe('TestAtom', () => {
        it('should match snapshot', () => {
            const wrapper = shalloWithTheme(<TestAtom filled key={1} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
