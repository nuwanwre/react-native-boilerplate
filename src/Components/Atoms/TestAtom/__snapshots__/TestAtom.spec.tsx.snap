// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components TestAtom should match snapshot 1`] = `
<Styled(TouchableOpacity)
  filled={true}
>
  <Text>
    Test Button
  </Text>
</Styled(TouchableOpacity)>
`;
